### ISSUE TITLE
@typescript-eslint/restrict-template-expressions reported in TS file due to imported type declared in a Vue component

### ISSUE THREAD

**Issue by TNGD-YQ** [vuejs/eslint-config-typescript#216]
# Problem

A type exported from a Vue component can be identified by VSCode properlym, but type aware linter complaints about it.

<img width="1567" height="1010" alt="Image" src="https://github.com/user-attachments/assets/147db586-5b8b-4afe-bb80-a89a9a408a03" />

# Reproduce Steps
1. Create a new project refer to https://vuejs.org/guide/quick-start#creating-a-vue-application.
2. In the `eslint.config.ts`, change `vueTsConfigs.recommended` to `vueTsConfigs.strictTypeChecked`
3. In the `App.vue`, export a type.

```vue
<script setup lang="ts">
export interface Item {
  value: string
  num: number
}
</script>
```

4. Create a file `test.ts`.

```ts
import type { Item } from '@/App.vue'

const a: Item = { value: 'test', num: 123 }

const str = `This is a template string: ${a.value} and the number is: ${a.num}`
```

# Expectation

`Item.value` is a string, so in `TS` file should not complainst about `any` type cannot be used in template literal expression.
